zng/docs/spec.md

```diff
@@ -468,17 +467,17 @@ A TZNG type binding has the following form:
 #<type-tag>:<type-string>
 ```
 Here, `<type-tag>` is a string decimal integer and `<type-string>`
-is a string defining a record type (`<rtype>`) according to the [TZNG type grammar](#42-type-grammar). They create
+is a string defining a type (`<type>`) according to the [TZNG type grammar](#42-type-grammar). They create
```

Comment:
@henridf: It was only in this review pass that I realized the full implications of what you'd helpfully added in https://github.com/brimsec/zq/pull/897/commits/0a39ecc53ff175df586705dcbcb8897b0d04d43a. While the reference to `<rtype>` (which I was in favor of at the time 😄 ) got the spec in sync with the `zq` implementation, being fully in sync between the two would also imply purging most of the examples from Section 4.4. After discussing with Steve & thinking about it myself I now recognize that would be going a little too far. So it still seems like your enhancements to the pseudo-grammar are very much helpful, but I _think_ relaxing this to just `<type>` makes the examples still legit (though still not implemented in `zq`). Am I correct about that?

Reply:
👍 Yup.

Reply:
One addition: the tzng grammar currently allows an empty record type (`record [ ]`). The implementation doesn't. We could call it out as a delta, or get rid of it altogether. I'd opt for the latter.

Reply:
In the binary ZNG context in [3.1.1.1 Record Typedef](https://github.com/brimsec/zq/blob/master/zng/docs/spec.md#3111-record-typedef), a closing remark also says:

> A record may not contain zero columns.

...which is effectively another way of saying "no empty records allowed", right? That would mean that as of this moment the TZNG grammar is out of sync with the implementation _and_ the binary ZNG spec, which seems to further validate your instincts that we should just remove `record [ ]` from the TZNG grammar, which I'll do in a commit now.